### ISSUE TITLE
Use model catalog api python client

### ISSUE THREAD

**Issue by mosoriob** [mintproject/wcm#28]
Why don't you use the model catalog api?

https://github.com/mintproject/model-catalog-python-api-client

https://github.com/mintproject/wcm/blob/fcd7b3d5822368319be2fadbb84709662f5b287b/src/wcm/_has_grid.py#L10

**Comment by dhruvp-8** [comment]
Actually, we had talked to @dgarijo regarding the use of model catalog python api client. Apparently I need to bind the data into class variables in the python api client in comparison to directly sending the data in JSON format using the requests package which is more convenient.

PS: We do validate the data before sending them in JSON using requests

**Comment by dgarijo** [comment]
Hi Dhruv,
Maybe I am missing something, but you should use the model catalog API, as
you are inserting information in the model catalog.
Daniel

**Comment by dhruvp-8** [comment]
Correct Daniel, we are using the model catalog API for inserting the data but we are not using model catalog python api package (https://github.com/mintproject/model-catalog-python-api-client) to do this, instead we are using requests package (as we can directly send the data in json format)

**Comment by mosoriob** [comment]
Since you don't use the client, you must handle the errors, configuration, and authentication. Also, you must maintain the code (new releases).

What are the benefits of a built-in solution?

**Comment by dgarijo** [comment]
I agree with @sirspock; why making your life more complicated? The code will be more difficult to maintain if we add new changes to the API.

**Comment by dhruvp-8** [comment]
Initially I was using the python client to do this stuff but the problem I was facing with that is I have to manually go and set the value of each property which is cumbersome. For eg: If I want to use Dataset Specification. I will have to manually insert the value of each property and if some properties are optional then also I have to set them to null(for string) or [] (For array). There are many properties inside a single object so setting each of their properties manually is not a good option. Whereas using requests we can directly send JSON (not set properties which are optional). Moreover, we perform the validation of YAML before sending the data so in that way there should not be any problems.

**Comment by maurya-rohit** [comment]
Just to add, we initially tried both methods and I think we consulted with @dgarijo if we can directly use the API, and you mentioned it is okay as python api client itself uses api in its core, it was for the user to publish any data. Moreover, it was convenient to use the api directly for following reason:
1. The variable names were same in yaml and api whereas we have to manually create an object for python api client where hasInput was declared as has_input. This added extra unnecessary effort.
2. Also, since we have to update the yaml file in to insert the ids for the newly generated data, it was easier to keep the whole data in json, since both are easily interchangeable. 
3. We are doing authentication when you configure mint-api. 
4. It was adding additional layer which was more difficult to maintain while we can directly use the api. As dhruv mentioned publishing and validation is different, so if there is any error it should be detected during the validation which was the point of creating two yaml so that we can validate easily.
